src/wp-includes/default-filters.php

```diff
@@ -595,6 +595,7 @@
 add_action( 'enqueue_block_assets', 'wp_enqueue_classic_theme_styles' );
 add_action( 'enqueue_block_assets', 'wp_enqueue_registered_block_scripts_and_styles' );
 add_action( 'enqueue_block_assets', 'enqueue_block_styles_assets', 30 );
+add_action( 'init', 'wp_load_classic_theme_block_styles_on_demand', 8 ); // Must happen before register_core_block_style_handles() at priority 9.
```

Comment:
I keep going back and forth on this, but why _unconditionally_ add the hook and _conditionally_ abort early if no block theme is present? why not group some of these into a big `if ( wp_is_block_theme() ) {}` block which then only adds the actions and filters if they are necessary?